src/main/java/com/mathworks/ci/MatlabBuildStepExecution.java

```diff
@@ -0,0 +1,107 @@
+    
+    private String tasks;
+
+    public MatlabBuildStepExecution(StepContext context, String tasks) {
+        super(context);
+        this.tasks = tasks;
+    }
+
+    private String getTasks() {
+        return this.tasks;
+    }
+
+    @Override
+    public Void run() throws Exception {
+        final Launcher launcher = getContext().get(Launcher.class);
+        final FilePath workspace = getContext().get(FilePath.class);
+        final TaskListener listener = getContext().get(TaskListener.class);
+        final EnvVars env = getContext().get(EnvVars.class);
+        
+        //Make sure the Workspace exists before run
+        
+        workspace.mkdirs();
+        
+        int exitCode = execMatlabCommand(workspace, launcher, listener, env);
+
+        if(exitCode != 0){
+            // throw an exception if return code is non-zero
+            stop(new MatlabExecutionException(exitCode));
+        }
+
+        getContext().setResult(Result.SUCCESS);
+        return null;
+    }
+
+    @Override
+    public void stop(Throwable cause) throws Exception {
+        getContext().onFailure(cause);
+    }
+    
+    private int execMatlabCommand(FilePath workspace, Launcher launcher,
+            TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
+        final String uniqueTmpFldrName = getUniqueNameForRunnerFile();  
+        final String uniqueBuildFile =
+                "build_" + getUniqueNameForRunnerFile().replaceAll("-", "_");
+        final FilePath uniqueTmpFolderPath =
+                getFilePathForUniqueFolder(launcher, uniqueTmpFldrName, workspace);
+
+        // Create MATLAB script
+        createMatlabScriptByName(uniqueTmpFolderPath, uniqueBuildFile, workspace, listener);
+        ProcStarter matlabLauncher;
+
+        try {
+            matlabLauncher = getProcessToRunMatlabCommand(workspace, launcher, listener, envVars,
+                    "cd('"+ uniqueTmpFolderPath.getRemote().replaceAll("'", "''") +"'); "+ uniqueBuildFile, uniqueTmpFldrName);
+            listener.getLogger()
+                    .println("#################### Starting command output ####################");
+            return matlabLauncher.pwd(workspace).join();
+
+        } finally {
+            // Cleanup the tmp directory
+            if (uniqueTmpFolderPath.exists()) {
+                uniqueTmpFolderPath.deleteRecursive();
+            }
+        }
+    }
+    
+    private void createMatlabScriptByName(FilePath uniqeTmpFolderPath, String uniqueScriptName, FilePath workspace, TaskListener listener) throws IOException, InterruptedException {
+
+        // Create a new command runner script in the temp folder.
+        final FilePath matlabBuildFile =
+                new FilePath(uniqeTmpFolderPath, uniqueScriptName + ".m");
+        final String tasks = getContext().get(EnvVars.class).expand(getTasks());
+        String cmd = "buildtool";
+
+        if (!tasks.trim().isEmpty()) {
```

Comment:
I don't love that the logic to make the MATLAB command is mixed with the logic to create script in the `createMatlabScriptByName` method